lib/plugin/instance.rb

```diff
@@ -247,7 +247,8 @@ def register_editable_group_custom_field(field)
   #     scope.where(word_count: value)
   #   end
   def register_custom_filter(filter_name, &block)
-    TopicsFilter.add_filter(filter_name, &block)
+    plugin = self
+    DiscoursePluginRegistry.register_custom_filter(filter_name, &block) if plugin.enabled?
```

Comment:
This `plugin.enabled?` check will only be run once, when the server boots (and for only one site in a multisite cluster).

Instead, we need to move it to be checked at runtime. DiscoursePluginRegistry has 'filtered registries' for that purpose. When adding things to the registry, you also pass in the plugin instance, so that it can check the enabled state when needed.

One example of where this is done with a `key => block` relationship would be `add_api_key_scope`